### PR TITLE
LPS-178419 [CP 2.0] Clearly display permanent (detached) activation keys along side annual subscriptions

### PR DESCRIPTION
**Pull request by kiwm**
Related Issue: https://issues.liferay.com/browse/LPS-171046

### CODE REVIEW

**Comment by kiwm** [comment]
ci:test:sf

**Comment by kiwm** [comment]
ci:test:sf

**Comment by kiwm** [comment]
ci:test:sf

**Comment by kiwm** [comment]
ci:test:relevant

**Comment by kiwm** [comment]
ci:test:relevant

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Project/Overview/components/SubscriptionsOverview/components/AccountSubscriptionsList/components/AccountSubscriptionModal/AccountSubscriptionModal.js [hunk 1]
```diff
@@ -11,16 +11,21 @@
 
 import ClayModal from '@clayui/modal';
 import {memo} from 'react';
+
 import i18n from '../../../../../../../../../../../common/I18n';
 import {
 	Button,
 	Table,
 } from '../../../../../../../../../../../common/components';
+import {useGetAccountSubscriptionUsage} from '../../../../../../../../../../../common/services/liferay/graphql/account-subscription-usage/queries/useGetAccountSubscriptionUsage';
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Project/Overview/components/SubscriptionsOverview/components/AccountSubscriptionsList/components/AccountSubscriptionModal/AccountSubscriptionModal.js:20]:
```suggestion
import {useGetAccountSubscriptionUsage} from '../../../../../../../../../../../common/services/liferay/graphql/account-subscription-usage';
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Project/Overview/components/SubscriptionsOverview/components/AccountSubscriptionsList/components/AccountSubscriptionModal/components/UsageGraph/UsageGraph.js
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Project/Overview/components/SubscriptionsOverview/components/AccountSubscriptionsList/components/AccountSubscriptionModal/components/UsageGraph/UsageGraph.js:1]: Hey @kiwm, you need to add the index.js file for this one too.

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Project/Overview/components/SubscriptionsOverview/components/AccountSubscriptionsList/components/AccountSubscriptionModal/AccountSubscriptionModal.js [hunk 2]
```diff
@@ -11,16 +11,21 @@
...
+import UsageGraph from './components/UsageGraph/UsageGraph';
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Project/Overview/components/SubscriptionsOverview/components/AccountSubscriptionsList/components/AccountSubscriptionModal/AccountSubscriptionModal.js:21]:
```suggestion
import UsageGraph from './components/UsageGraph';
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Project/Overview/components/SubscriptionsOverview/components/AccountSubscriptionsList/components/AccountSubscriptionModal/AccountSubscriptionModal.js [hunk 3]
```diff
@@ -33,6 +38,11 @@ const AccountSubscriptionModal = ({
 		{data, loading},
 	] = useOrderItems(externalReferenceCode);
 
+	const accountSubscriptionUsage = useGetAccountSubscriptionUsage(
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Project/Overview/components/SubscriptionsOverview/components/AccountSubscriptionsList/components/AccountSubscriptionModal/AccountSubscriptionModal.js:41]:
```suggestion
	const {data: accountSubscriptionUsageData, loading: accountSubscriptionUsageLoading } = useGetAccountSubscriptionUsage(
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Project/Overview/components/SubscriptionsOverview/components/AccountSubscriptionsList/components/AccountSubscriptionModal/AccountSubscriptionModal.js [hunk 4]
```diff
@@ -55,6 +65,18 @@ const AccountSubscriptionModal = ({
...
+					{i18n.translate('active-subscriptions')}
+				</h5>
+
+				{!accountSubscriptionUsage.loading && (
+					<UsageGraph
+						accountSubscriptionUsageData={
+							accountSubscriptionUsage.data
+						}
+					/>
+				)}
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Project/Overview/components/SubscriptionsOverview/components/AccountSubscriptionsList/components/AccountSubscriptionModal/AccountSubscriptionModal.js:78]:
```suggestion
                                       <UsageGraph
						data={
							accountSubscriptionUsageData?.getAccountSubscriptionUsage
						}
						loading={accountSubscriptionUsageLoading}
					/>
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Project/Overview/components/SubscriptionsOverview/components/AccountSubscriptionsList/components/AccountSubscriptionModal/components/UsageGraph/UsageGraph.js [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+	CartesianGrid,
+	Legend,
+	Tooltip,
+	XAxis,
+	YAxis,
+} from 'recharts';
+import i18n from '../../../../../../../../../../../../../common/I18n';
+import {Skeleton} from '../../../../../../../../../../../../../common/components';
+
+const UsageGraph = ({accountSubscriptionUsageData}) => {
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Project/Overview/components/SubscriptionsOverview/components/AccountSubscriptionsList/components/AccountSubscriptionModal/components/UsageGraph/UsageGraph.js:23]:
```suggestion
const UsageGraph = ({data, loading}) => {
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Project/Overview/components/SubscriptionsOverview/components/AccountSubscriptionsList/components/AccountSubscriptionModal/components/UsageGraph/UsageGraph.js [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+	if (!accountSubscriptionUsageData.getAccountSubscriptionUsage) {
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Project/Overview/components/SubscriptionsOverview/components/AccountSubscriptionsList/components/AccountSubscriptionModal/components/UsageGraph/UsageGraph.js:24]:
```suggestion
	if (loading || !data) {
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Project/Overview/components/SubscriptionsOverview/components/AccountSubscriptionsList/components/AccountSubscriptionModal/components/UsageGraph/UsageGraph.js [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+		return (
+			<Skeleton align="center" className="mb-5" height={20} width={100} />
+		);
+	}
+
+	const {
+		getAccountSubscriptionUsage: {annualSubscriptions, currentConsumption},
+	} = accountSubscriptionUsageData;
+
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Project/Overview/components/SubscriptionsOverview/components/AccountSubscriptionsList/components/AccountSubscriptionModal/components/UsageGraph/UsageGraph.js:33]:
```suggestion
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Project/Overview/components/SubscriptionsOverview/components/AccountSubscriptionsList/components/AccountSubscriptionModal/components/UsageGraph/UsageGraph.js [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+	return (
+		<div>
+			<div className="d-flex justify-content-center">
+				<BarChart
+					cursor="pointer"
+					data={annualSubscriptions}
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Project/Overview/components/SubscriptionsOverview/components/AccountSubscriptionsList/components/AccountSubscriptionModal/components/UsageGraph/UsageGraph.js:39]:
```suggestion
					data={data.annualSubscriptions}
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Project/Overview/components/SubscriptionsOverview/components/AccountSubscriptionsList/components/AccountSubscriptionModal/components/UsageGraph/UsageGraph.js [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+						stackId="maxConcurrentQuantity"
+					/>
+				</BarChart>
+			</div>
+
+			<div className="d-flex flex-row-reverse m-2 mr-3">
+				<h6>
+					{i18n.translate('keys-provisioned-total') +
+						': ' +
+						currentConsumption}
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Project/Overview/components/SubscriptionsOverview/components/AccountSubscriptionsList/components/AccountSubscriptionModal/components/UsageGraph/UsageGraph.js:104]:
```suggestion
						data.currentConsumption}
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Project/Overview/components/SubscriptionsOverview/components/AccountSubscriptionsList/components/AccountSubscriptionModal/components/UsageGraph/UsageGraph.js [hunk 10]
```diff
@@ -0,0 +1,110 @@
...
+				</h6>
+			</div>
+		</div>
+	);
+};
+export default UsageGraph;
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Project/Overview/components/SubscriptionsOverview/components/AccountSubscriptionsList/components/AccountSubscriptionModal/components/UsageGraph/UsageGraph.js:110]:
```suggestion
};

export default UsageGraph;
```

**Review by jairmedeiros — CHANGES_REQUESTED** [review]

**Comment by kiwm** [comment]
ci:test:sf

**Comment by kiwm** [comment]
ci:test:sf

**Comment by kiwm** [comment]
ci:test:relevant

**Review by JoMaAlves — APPROVED** [review]
LGTM ​🦖​🌋​ 
Thank you @kiwm

**Review by jairmedeiros — APPROVED** [review]

**Comment by kiwm** [comment]
ci:test:sf

**Review by JoMaAlves — APPROVED** [review]
LGTM ​🦖​🌋​ 
Thank you @kiwm

**Comment by JoMaAlves** [comment]
ci:forward

**Comment by jairmedeiros** [comment]
ci:forward

**Pull request closed without merging**